UndertaleModLib/Decompiler/Decompiler.cs

```diff
@@ -1597,10 +1776,13 @@ private static List<Statement> HLDecompile(Dictionary<uint, Block> blocks, Block
 
             // TODO: This doesn't propagate to tempvars in broken switch statements but that will be cleaned up at some point
             AssetIDType propType = AssetIDType.Other;
-            if (code.Name.Content == "gml_Script___view_set_internal" || code.Name.Content == "gml_Script___view_get")
-                propType = AssetIDType.e__VW;
-            if (code.Name.Content == "gml_Script___background_set_internal" || code.Name.Content == "gml_Script___background_get_internal")
-                propType = AssetIDType.e__BG;
+            // NOTE: This was disabled because it wasn't valid.
+            // 1. The type only applied to the first case statement.
+            // 2. Even if each case statement had the proper type, it still gives an error when compiling.
```

Comment:
It throws an error because the enum needs to be defined in a script, see `C:\ProgramData\GameMakerStudio2\Cache\runtimes\runtime-X.X.X.XXX\lib\compatibility.zip`, specifically `background\__init_background.gml` and `view\__init_view.gml`

When going for full decompilation you may want to copy the scripts from there (if GMS2 is installed) instead of decompiling them since they have proper comments etc. - although it is technically possible for somebody to modify them...

And type propagation in the case statement is another problem, it happens because switch statements are resolved after type propagation so during type propagation you have some tempvar nonsense going on

Reply:
Good to note. Hmm. I think if we want to expect someone to grab stuff from a compatibility zip file, we should tell them somehow, if not grab it ourselves.